### PR TITLE
Generic substrate app(Adds Libarland)

### PR DESCRIPTION
**Pull request by N1ghtStorm**

### CODE REVIEW

File: pallets/sora-liberland-bridgeproxy/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
```
Comment (Alexey-N-Chernyshov) [inline pallets/sora-liberland-bridgeproxy/src/lib.rs]: It is not our header. Should it be replaced?

File: pallets/substrate-app/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,687 @@
...
+            GenericBalance,
+        >;
+
+        type BridgeAssetLocker: BridgeAssetLocker<Self::AccountId>;
+
+        type WeightInfo: WeightInfo;
+    }
+
+    #[pallet::hooks]
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
```
Comment (Alexey-N-Chernyshov) [inline pallets/substrate-app/src/lib.rs]: Do you need this if there is no hooks?

File: pallets/substrate-app/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,687 @@
...
+            BalanceOf<T>,
+        ),
+        /// [network_id, asset_id, sender, recepient, amount]
+        Minted(
+            SubNetworkId,
+            AssetIdOf<T>,
+            GenericAccount,
+            T::AccountId,
+            BalanceOf<T>,
+        ),
```
Comment (Alexey-N-Chernyshov) [inline pallets/substrate-app/src/lib.rs]: I think name fields are more concise:
```suggestion
        Burned(
            network_id: SubNetworkId,
            asset_id: AssetIdOf<T>,
            sender: T::AccountId,
            recipient: GenericAccount,
            amount: BalanceOf<T>,
        ),
        Minted(
            network_id: SubNetworkId,
            asset_id: AssetIdOf<T>,
            sender: GenericAccount,
            recipient: T::AccountId,
            amount: BalanceOf<T>,
        ),
```

File: pallets/sora-liberland-bridgeproxy/src/lib.rs (hunk 1 above)
Comment (N1ghtStorm) [reply pallets/sora-liberland-bridgeproxy/src/lib.rs]: Done

File: pallets/substrate-app/src/lib.rs (hunk 2 above)
Comment (N1ghtStorm) [reply pallets/substrate-app/src/lib.rs]: removed

File: pallets/substrate-app/src/lib.rs (hunk 3 above)
Comment (N1ghtStorm) [reply pallets/substrate-app/src/lib.rs]: Are you sure that that should work?

File: pallets/liberland-bridge-provider/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,488 @@
...
+                match asset_kind {
+                    bridge_types::types::AssetKind::Thischain => {
+                        T::Balances::transfer(
+                            who,
+                            &tech_acc,
+                            (*amount).into(),
+                            ExistenceRequirement::AllowDeath,
+                        )?;
+                    },
+                    bridge_types::types::AssetKind::Sidechain => {
```
Comment (vovac12) [inline pallets/liberland-bridge-provider/src/lib.rs]: Sidechain LLD?

File: pallets/substrate-app/src/lib.rs (hunk 3 above)
Comment (vovac12) [reply pallets/substrate-app/src/lib.rs]: It will work with struct-like variants. For example `Burned { ... }`

**Review by vovac12 — APPROVED** [review]

**Pull request merged**